src/Integrations/Integrations/CakePHP/CakePHPIntegration.PHP5.php

```diff
@@ -58,13 +58,18 @@ public function init()
                 $integration->rootSpan->meta[Tag::SPAN_KIND] = 'server';
             }
 
+            $integration->rootSpan->meta[Tag::SPAN_KIND] = 'server';
+            $integration->rootSpan->meta[Tag::COMPONENT] = $this->getName();
+
             \DDTrace\trace_method(
                 'Controller',
                 'invokeAction',
                 function (SpanData $span, array $args) use ($integration) {
                     $span->name = $span->resource = 'Controller.invokeAction';
                     $span->type = Type::WEB_SERVLET;
                     $span->service = $integration->appName;
+                    $span->meta[Tag::SPAN_KIND] = 'server';
+                    $span->meta[Tag::COMPONENT] = $this->getName();
```

Comment:
Inside of closures that are being passed to `trace_method`, the `$this` variable will get rebound to the object it's attached to (for this specifica case, the `Controller`. Pair that with the other comment about using `NAME` instead, and we get:

```suggestion
                    $span->meta[Tag::COMPONENT] = $integration::NAME;
```

This applies to many other places too.